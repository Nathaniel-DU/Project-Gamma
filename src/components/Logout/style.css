button#logout-button {
    background-color: #00203FFF;
    margin: auto;
    min-width: 90%;
    cursor: pointer;
    margin-right: 0.25em;
    margin-top: 0.5em;
    /* padding: 	0.938em; */
    border: none;
    border-radius: 10px;
    color: #fefefe;
    margin-bottom: 10px;
}

button:hover {
    background-color: #4A4E69!important;
    color: #fefefe!important;
  }  