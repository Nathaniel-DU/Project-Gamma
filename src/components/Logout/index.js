import React, {Component} from 'react';
import axios from 'axios';
import './style.css';

export default class Logout extends Component {

    logOut = () => {
        axios.put(`/event/stop`)
            .then(() => {
                axios.get(`/auth/logout`)
                    .then(res => {
                        if(res.status === 200){
                            window.location.replace(`/`);
                        }
                    });
            });
    }
    
    render(){
        return(
            <div>
                <button className='blue-button' id='logout-button' onClick={this.logOut}>Logout</button>
            </div>
        );
    }
}
