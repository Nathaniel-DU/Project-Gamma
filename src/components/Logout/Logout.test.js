import React from 'react';
import { shallow } from 'enzyme';
import Logout from './index';

describe(`Logout`, () => {
    it(`should render correctly in "debug" mode`, () => {
        const component = shallow(<Logout debug />);
        expect(component).toMatchSnapshot();
    });
});