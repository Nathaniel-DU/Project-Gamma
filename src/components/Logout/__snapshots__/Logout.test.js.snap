// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should render correctly in "debug" mode 1`] = `
<div>
  <Link
    to="/"
  >
    <button
      className="blue-button"
      id="logout-button"
      onClick={[Function]}
    >
      Logout
    </button>
  </Link>
</div>
`;
